Source/ASRunLoopQueue.mm

```diff
@@ -196,8 +200,58 @@ @interface ASRunLoopQueue () {
 
 @end
 
+#if AS_KDEBUG_ENABLE
+typedef enum {
+  kCATransactionPhasePreLayout,
+  kCATransactionPhasePreCommit,
+  kCATransactionPhasePostCommit,
+} CATransactionPhase;
+
+@interface CATransaction (Private)
++ (void)addCommitHandler:(void(^)(void))block forPhase:(CATransactionPhase)phase;
```

Comment:
Interesting, is this a real API? I didn't see it implemented in this patch. Given the compile-time gating, seems like a great thing to use.

Even though debug-only, since this could change in any iOS version (or maybe even doesn't exist on one of the older versions), it's worth adding an early-return to the registerObservers method checking [CALayer respondsToSelector:].